src/chain/stateTransition/index.ts

```diff
@@ -21,14 +21,21 @@ export {
 export function executeStateTransition(state: BeaconState, block: BeaconBlock | null): BeaconState {
   cacheState(state);
   advanceSlot(state);
+  if (shouldProcessEpoch(state)) {
+    processEpoch(state);
+  }
 
   if (block) {
+    while(state.slot < block.slot) {
+      cacheState(state);
+      advanceSlot(state);
+      if (shouldProcessEpoch(state)) {
+        processEpoch(state);
+      }
+    }
```

Comment:
I think we will want to process an epoch even if there is a null block

Reply:
We are, check above if statement, if there is a block we just do a slot and epoch catchup